src/bootstrap/bootstrap.py

```diff
@@ -378,13 +380,28 @@ def download_stage0(self):
             with output(self.cargo_stamp()) as cargo_stamp:
                 cargo_stamp.write(self.date)
 
-    def _download_stage0_helper(self, filename, pattern):
+        if self.rustfmt().startswith(self.bin_root()) and (
+            not os.path.exists(self.rustfmt())
+            or self.program_out_of_date(self.rustfmt_stamp())
+        ):
+            if rustfmt_channel:
```

Comment:
hm, I wonder if it makes sense to check if rustfmt_channel *first* and then go through the bin_root etc checks? In particular I worry that program_out_of_date and such may eventually error out if the channel doesn't exist, for whatever reason